### ISSUE TITLE
Global macros not working in modals

### ISSUE THREAD

**Issue by felixhayashi** [Jermolene/TiddlyWiki5#1472]
Hi @Jermolene,

Please try the following to verify the issue:

t1 containing the trigger

    title: t1

    <$button>Click me!
      <$action-sendmessage $message="tm-modal" $param="t3" />
    </$button>
    
t2 containing the macro

    title: t2
    tags: $:/tags/Macro
    
    \define mysamplemacro(name:"Bugs Bunny",address:"Rabbit Hole Hill")
    Hi, I'm $name$ and I live in $address$
    \end

t3 invoking the macro

    title: t3
    
    <<mysamplemacro>>

This works when not used in a modal...

Regards
Felix

**Comment by Jermolene** [comment]
Hi @felixhayashi that's correct. Modals are created as independent widget render trees, and don't inherit the global macros that are included in the main page template.

A workaround that you can sometimes use is to include the `<$importvariables>` call from the main page template in your modal. 

Also note that you can pass variables to the modal as described in "WidgetMessage: tm-modal".

**Comment by felixhayashi** [comment]
Hi @Jermolene,

big thanks for your help here!

`<$importvariables>` is such a great piece of code! It helped and also answered my question posted in the google groups (https://groups.google.com/d/msg/tiddlywiki/KGPqvYpJVH0/EPwmqzqfpAkJ).

> Also note that you can pass variables to the modal as described in "WidgetMessage: tm-modal".

Yes, I remember we worked hard on that one :) In fact my modals became so complex, that, I added a library that is only concerned with managing the modals (instantiating temporary templates; injecting predefining fields; listening for confirm or success events, parsing the result, calling callbacks, cleaning up; ...see https://github.com/felixhayashi/TW5-TiddlyMap/blob/master/src/plugins/felixhayashi/tiddlymap/lib.dialog_manager.js).

-Felix
